libraries/arrow-convert/src/from_impls.rs

```diff
@@ -18,6 +18,18 @@ impl From<arrow::array::ArrayRef> for ArrowData {
     }
 }
 
+impl From<ArrowData> for arrow::array::ArrayData {
+    fn from(value: ArrowData) -> Self {
+        value.0.to_data()
+    }
+}
```

Comment:
`ArrowData` implements `Deref` to `ArrayRef`. So you should be able to just call `to_data` on the `ArrowData` instance directly. (We also provide a `impl From<ArrowData> for arrow::array::ArrayRef` implementation, so if you need ownership of the underlying `ArrayRef` instance.)

Given that the official `arrow` crate doesn't provide a `From` conversion from `ArrayRef` to `ArrayData`, I'm not sure whether we should provide such an implementation.

Reply:
Make sense, sure!